src/main/java/org/java_websocket/WebSocketImpl.java

```diff
@@ -410,6 +412,14 @@ private void decodeFrames(ByteBuffer socketBuffer) {
       log.error("Closing due to invalid data in frame", e);
       wsl.onWebsocketError(this, e);
       close(e);
+    } catch (VirtualMachineError | ThreadDeath | LinkageError e) {
+      log.error("Got fatal error during frame processing");
+      throw e;
+    } catch (Error e) {
+      log.error("Closing web socket due to an error during frame processing");
+      Exception exception = new Exception(e);
+      wsl.onWebsocketError(this, exception);
+      close(CloseFrame.UNEXPECTED_CONDITION, exception.getMessage());
```

Comment:
I'm not sure if its a good idea to send exception message to the client. There are a lot of subclasses of Error which could leak important information. I've personally seen cases where data like string lengths, file paths and sql queries is leaked through exception messages.

Reply:
@PhilipRoman yeap good hint, changed to just return type of error without any additional information

Reply:
@Serpion-ua Looks good!
One last thing, the new error message mentions "server side", but WebSocketImpl is used for both server and client so you should probably remove that part.

Reply:
@PhilipRoman yeah removed that part.